### PR TITLE
Update to compose release 0.1.0-dev08

### PR DESCRIPTION
**Pull request by PranavMaganti**
Grade -> 4.1.0-alpha04
Ripple, Padding, LayoutWidth, LayoutHeight, expanded -> respective Modifiers
Container -> Box

### CODE REVIEW

**Comment by zsoltk** [comment]
@vanpra Thanks for the PR!

I noticed a lot of things are broken visually in both example apps. Can you please compare it to previous version. Most notably text colours make many texts blend in the background in new version (completely invisible), button text colors became black, photo gallery photo full screen disappeared. Please test the apps to make sure they look and behave the same as before.

I'm not sure any of the above is a mistake of migrating text styles, or has anything to do with gradle changes that don't reflect the updates suggested in, but you might want to check: https://developer.android.com/jetpack/androidx/releases/compose

File: app-lifelike/src/main/java/com/example/lifelike/composable/LoggedIn.kt [hunk 1]
```diff
@@ -26,9 +28,9 @@ interface LoggedIn {
                 val routing = backStack.last()
 
                 Column {
-                    Container(
-                        modifier = LayoutWeight(1f),
-                        alignment = Alignment.TopStart
+                    Box(
+                        modifier = Modifier.weight(1f)
+                                + Modifier.wrapContentSize(Alignment.TopStart)
```
Comment (zsoltk) [inline app-lifelike/src/main/java/com/example/lifelike/composable/LoggedIn.kt]: I believe the suggested new style is chaning the modifiers with fluent API.

File: app-lifelike/src/main/java/com/example/lifelike/composable/LoggedOut.kt [hunk 2]
```diff
@@ -1,11 +1,7 @@
 package com.example.lifelike.composable
 
 import androidx.compose.Composable
-import com.example.lifelike.composable.loggedout.RegConfirmSmsCode
-import com.example.lifelike.composable.loggedout.RegFinal
-import com.example.lifelike.composable.loggedout.RegUserName
-import com.example.lifelike.composable.loggedout.RegUserPhone
-import com.example.lifelike.composable.loggedout.Splash
+import com.example.lifelike.composable.loggedout.*
```
Comment (zsoltk) [inline app-lifelike/src/main/java/com/example/lifelike/composable/LoggedOut.kt]: No star imports please.

File: app-lifelike/src/main/java/com/example/lifelike/composable/LoggedOut.kt [hunk 3]
```diff
@@ -39,8 +35,14 @@ interface LoggedOut {
...
-                    Routing.RegUserName -> RegUserName.Content(user = user, onNext = currentRouting.next())
-                    Routing.RegUserPhone -> RegUserPhone.Content(user = user, onNext = currentRouting.next())
+                    Routing.RegUserName -> RegUserName.Content(
+                        user = user,
+                        onNext = currentRouting.next()
+                    )
+                    Routing.RegUserPhone -> RegUserPhone.Content(
+                        user = user,
+                        onNext = currentRouting.next()
+                    )
```
Comment (zsoltk) [inline app-lifelike/src/main/java/com/example/lifelike/composable/LoggedOut.kt:49]: I'd like to keep code formatting in most cases as it was before. I agree with some of the code style modifications, but not all. The PR should ideally introduce changes only in its scope (migrating to dev08).

File: app-lifelike/src/main/java/com/example/lifelike/composable/loggedin/FullScreenPhoto.kt [hunk 4]
```diff
@@ -1,14 +1,12 @@
...
+import androidx.ui.foundation.Box
+import androidx.ui.foundation.Text
 import androidx.ui.graphics.Color
-import androidx.ui.layout.Column
-import androidx.ui.layout.Container
-import androidx.ui.layout.LayoutHeight
-import androidx.ui.layout.LayoutPadding
-import androidx.ui.layout.Spacer
+import androidx.ui.graphics.ImageShader
+import androidx.ui.layout.*
```
Comment (zsoltk) [inline app-lifelike/src/main/java/com/example/lifelike/composable/loggedin/FullScreenPhoto.kt]: No star imports please.

File: app-lifelike/src/main/java/com/example/lifelike/composable/loggedin/FullScreenPhoto.kt [hunk 5]
```diff
@@ -23,16 +21,16 @@ interface FullScreenPhoto {
...
-            val typography = MaterialTheme.typography()
+            val typography = MaterialTheme.typography
 
             Surface(color = Color.DarkGray) {
-                Container(modifier = LayoutPadding(32.dp)) {
+                Box(modifier = Modifier.padding(32.dp)) {
                     Column {
-                        Container(modifier = LayoutWeight(0.8f), expanded = true) {
-                            Image(image = image)
+                        Box(modifier = Modifier.weight(0.8f) + Modifier.fillMaxSize()) {
```
Comment (zsoltk) [inline app-lifelike/src/main/java/com/example/lifelike/composable/loggedin/FullScreenPhoto.kt]: Let's use fluent api for modifiers here

File: app-lifelike/src/main/java/com/example/lifelike/composable/loggedin/Menu.kt [hunk 6]
```diff
@@ -61,7 +62,7 @@ interface Menu {
                     onClick = { onClick.invoke(item) }
                 ) {
                     Text(
-                        modifier = LayoutWidth.Fill + LayoutPadding(8.dp),
+                        modifier = Modifier.fillMaxWidth() + Modifier.padding(8.dp),
```
Comment (zsoltk) [inline app-lifelike/src/main/java/com/example/lifelike/composable/loggedin/Menu.kt]: Let's use fluent api for modifiers here

File: app-lifelike/src/main/java/com/example/lifelike/composable/loggedin/PhotosOfAlbum.kt [hunk 7]
```diff
@@ -1,16 +1,9 @@
...
-import androidx.ui.foundation.VerticalScroller
-import androidx.ui.layout.Column
-import androidx.ui.layout.Container
-import androidx.ui.layout.LayoutAspectRatio
-import androidx.ui.layout.LayoutPadding
-import androidx.ui.layout.Table
-import androidx.ui.layout.TableColumnWidth
+import androidx.ui.core.Modifier
+import androidx.ui.foundation.*
+import androidx.ui.layout.*
```
Comment (zsoltk) [inline app-lifelike/src/main/java/com/example/lifelike/composable/loggedin/PhotosOfAlbum.kt]: No star imports please.

File: app-lifelike/src/main/java/com/example/lifelike/composable/loggedin/Profile.kt [hunk 8]
```diff
@@ -20,22 +17,22 @@ interface Profile {
         @Composable
         fun Content(user: User, onLogout: () -> Unit) {
             Column(
-                modifier = LayoutHeight.Fill + LayoutPadding(40.dp),
+                modifier = Modifier.fillMaxHeight() + Modifier.padding(40.dp),
```
Comment (zsoltk) [inline app-lifelike/src/main/java/com/example/lifelike/composable/loggedin/Profile.kt]: Let's use fluent api for modifiers here.

File: app-lifelike/src/main/java/com/example/lifelike/composable/loggedout/Splash.kt [hunk 9]
```diff
@@ -21,20 +19,20 @@ interface Splash {
...
+                    style = MaterialTheme.typography.h4
                 )
-                Container(
-                    expanded = true,
-                    constraints = DpConstraints(maxHeight = 48.dp),
-                    alignment = Alignment.Center
+                Box(
+                    modifier = Modifier.fillMaxSize()
+                            + Modifier.preferredSizeIn(DpConstraints(maxHeight = 48.dp))
+                            + Modifier.wrapContentSize(Alignment.Center)
```
Comment (zsoltk) [inline app-lifelike/src/main/java/com/example/lifelike/composable/loggedout/Splash.kt]: Let's use fluent api for modifiers here
```suggestion
                    modifier = Modifier
                            .fillMaxSize()
                            .preferredSizeIn(DpConstraints(maxHeight = 48.dp))
                            .wrapContentSize(Alignment.Center)
```

File: app-lifelike/src/main/java/com/example/lifelike/composable/loggedout/common/RegFlowPanel.kt [hunk 10]
```diff
@@ -2,12 +2,10 @@ package com.example.lifelike.composable.loggedout.common
...
-import androidx.ui.core.Text
-import androidx.ui.layout.Arrangement
-import androidx.ui.layout.Column
-import androidx.ui.layout.Container
-import androidx.ui.layout.DpConstraints
-import androidx.ui.layout.LayoutPadding
+import androidx.ui.core.Modifier
+import androidx.ui.foundation.Box
+import androidx.ui.foundation.Text
+import androidx.ui.layout.*
```
Comment (zsoltk) [inline app-lifelike/src/main/java/com/example/lifelike/composable/loggedout/common/RegFlowPanel.kt]: No star import please.

File: app-lifelike/src/main/java/com/example/lifelike/composable/loggedout/common/RegFlowPanel.kt [hunk 11]
```diff
@@ -18,16 +16,16 @@ fun RegFlowPanel(
...
         )
         content()
-        Container(
-            expanded = true,
-            constraints = DpConstraints(maxHeight = 48.dp),
-            alignment = Alignment.CenterEnd
+        Box(
+            modifier = Modifier.fillMaxSize()
+                    + Modifier.preferredSizeIn(DpConstraints(maxHeight = 48.dp))
+                    + Modifier.wrapContentSize(Alignment.CenterEnd)
```
Comment (zsoltk) [inline app-lifelike/src/main/java/com/example/lifelike/composable/loggedout/common/RegFlowPanel.kt]:
```suggestion
            modifier = Modifier
                    .fillMaxSize()
                    .preferredSizeIn(DpConstraints(maxHeight = 48.dp))
                    .wrapContentSize(Alignment.CenterEnd)
```

**Review by zsoltk — CHANGES_REQUESTED** [review]

**Comment by PranavMaganti** [comment]
Thanks for the comments. This should fix most of the stuff you mentioned. I still have to take a look at the layout issues. Looks like the new modifiers have messed a bit with the layout and I'm still investigating the Text colour issue. Also sorry about the formatting issues, I'm just used to auto formatting all the code with my settings.

**Comment by PranavMaganti** [comment]
That should be everything fixed now

**Comment by zsoltk** [comment]
@vanpra Hey, did you actually check it? Both example apps are still broken. Can you please compare against `0.5.2` and have at least the disappearing things back on screen again.

I played around on top of your latest commit and one of the issues seems to be that `Column` and `Row` no longer seem to work with "naked" elements. Once I added `Box`es around everything (as I found in [ColumnSample](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-master-dev:ui/ui-layout/samples/src/main/java/androidx/ui/layout/samples/ColumnSample.kt;l=39?q=SimpleColumn&ss=androidx%2Fplatform%2Fframeworks%2Fsupport)) it started working again. 

This is needed in all places containing `Column` or `Row`.

Also probably some more button text colors need tweeking in nested containers, but I'd be happy if that was the only issue.

Do you want to fix these?

**Comment by PranavMaganti** [comment]
@zsoltk I've had another look to make sure and the apps look exactly the same except for a slight alignment issue on the logged out screens, everything seems to be visible. Could you please specify the screens which you saw these issues on. I've also had a look at the nested containers app and both apps look identical.

**Comment by zsoltk** [comment]
@vanpra 

1. `app-nested-containers`: back stack count and local counters are not visible. Only the purple buttons are. On API 28 the padding is 0 for them, while strangely in API 25 that's fine.

2. `app-example`: On API 28 the reg flow panels are collapsed to the top and buttons are not even clickable. Profile page after login is also broken. On API 25 it's looking fine.

The first one is broken regardless of API level. The second example doesn't look exactly your fault since there's API level differences in behaviour. Might be worth asking about it on Slack. 

Nonetheless, if you wrap Composables into `Box` elements instead of putting them directly under `Column` and `Row`, then all the above problematic cases start working correctly.

**Comment by zsoltk** [comment]
@vanpra Btw created #17 to make your life easier (as well as future Compose version updates). It's up to you which way you want to go now in this PR: you can either fix the issues, or you can make the example apps point to an earlier version, just tell me how you would like to proceed.

**Comment by PranavMaganti** [comment]
I've had a quick look at the issue for the app-nested-containers and I've wrapped children of a column or row in a box but for some reason the text still seems to be invisible

**Pull request merged**